test/common_nn.py

```diff
@@ -322,11 +322,55 @@ def smoothl1loss_reference(input, target, size_average=True, reduce=True):
     return output
 
 
+def _multilabelmarginloss_reference(input, target, is_target):
+    sum = 0
+    for i in range(0, target.size()[0]):
+        target_index = target[i]
```

Comment:
It's a little hard to understand this (I understand that the THNN implementation uses something similar). Is it possible to rewrite this to look more like the formula? Maybe capture the `i != y[j]` part without using an `is_target` array? It's okay if it's not as efficient since we want this function to only be used in testing.